shuffle-plugin/src/main/scala/com/nvidia/spark/rapids/shuffle/ucx/UCX.scala

```diff
@@ -69,7 +69,8 @@ class UCX(executorId: Int, usingWakeupFeature: Boolean = true) extends AutoClose
 
   private var worker: UcpWorker = _
   private val endpoints = new ConcurrentHashMap[Long, UcpEndpoint]()
-  private var initialized = false
+  @volatile private var initialized = false
+  private[this] val initializedMonitor = new Object
```

Comment:
It would be useful to have some comments explaining why a separate monitor object is required rather than locing the `UCX` instance directly.

Reply:
It isn't required to have a separate monitor. Locking the instance should be fine here, let me rework.

Reply:
done.